text/0202-events-and-logs-api.md

```diff
@@ -0,0 +1,193 @@
+  - In Python, logging.LogRecord's extra field is mapped to Otel LogRecord's attributes but this field is a hidden field and not part of the public interface.
+- The current Log APIs have a message parameter which could map to the Body field of LogRecord. However, this is restricted to String messages and does not allow for structured logs.
+
+For the above reasons we can conclude that we will need an API for generating Events API
+
+## Should OpenTelemetry have an API for logs?
+
+A part of the OTel community thnks that we should not have a full-fledged logging API unless there is a language that doesn't already have a plethora of logging libraries & APIs to choose from where it might make sense to define one. Further, we will not be able to have the [rich set of configuration options](https://logging.apache.org/log4j/2.x/manual/configuration.html) that some popular logging frameworks provide so the logging API in OTel will only become yet another API. However, it was noted that the Log Appender API is very similar to the API for Events and so instead of having API for Events and API for Log Appenders separately it was agreed to have one API for Events and Logs, and that the API for Logs is targeted only to Log Appenders.
+
+## Events and Logs API Interface
+
+For reference, a prototype of the Events and Logs API in Java is [here](https://github.com/scheler/opentelemetry-java/pull/1/files)
+
+Client-side telemetry is one of the initial clients that will use the Events API and so the API will be made available in JavaScript, Java and Swift first to be able to use in the SDKs for Browser, Android and iOS.  It may also be added in Go since there is a Kubernetes events receiver implemented in Collector based on Logs data-model.
+
+The Events and Logs API consist of these main classes:
+
+* LoggerProvider is the entry point of the API. It provides access to Loggers.
+* Logger is the class responsible for creating events using Log records.
+
+![Events and Logs API classes](img/0202-events-and-logs-api.png)
+
+### LoggerProvider
+
+Logger can be accessed with an LoggerProvider.
+
+In implementations of the API, the LoggerProvider is expected to be the stateful object that holds any configuration.
+
+Normally, the LoggerProvider is expected to be accessed from a central place. Thus, the API SHOULD provide a way to set/register and access a global default LoggerProvider.
+
+Notwithstanding any global LoggerProvider, some applications may want to or have to use multiple LoggerProvider instances, e.g. to have different configuration (like LogRecordProcessors) for each (and consequently for the Loggers obtained from them), or because it's easier with dependency injection frameworks. Thus, implementations of LoggerProvider SHOULD allow creating an arbitrary number of Logger instances.
```

Comment:
```suggestion
Notwithstanding any global LoggerProvider, some applications may want to or have to use multiple LoggerProvider instances, e.g. to have different configuration (like LogRecordProcessors) for each (and consequently for the Loggers obtained from them), or because it's easier with dependency injection frameworks. Thus, implementations of LoggerProvider SHOULD allow creating an arbitrary number of instances.
```